fast64_internal/f3d/f3d_material.py

```diff
@@ -416,7 +488,24 @@ def ui_geo_mode(settings, dataHolder, layout, useDropdown):
     if not useDropdown or dataHolder.menu_geo:
         disable_dependent = False  # Don't disable dependent props in world defaults
 
+        def should_draw(*props):
```

Comment:
I'm gonna let Python experts decide if this is okay or not, I think it looks weird but maybe it's fine idk, also fwiw I'm not a fan of nested functions like this

Reply:
yeah I really want some opinions on this part of the pr especially

Reply:
I like how readable it is